### PR TITLE
Fixing Gruntfile

### PR DESCRIPTION
**Pull request by marktheunissen**
Grunt file changes as discussed with @elliotttf in the car this morning.

### CODE REVIEW

File: controller/web/Gruntfile.js [hunk 1]
```diff
@@ -13,101 +13,117 @@ module.exports = function(grunt) {
...
-    // This task uses the cssmin Node.js project to take all your CSS files in
-    // order and concatenate them into a single CSS file named index.css.  It
-    // also minifies all the CSS as well.  This is named index.css, because we
-    // only want to load one stylesheet in index.html.
     cssmin: {
-      'dist/release/index.css': [
-        'css/alertify.css',
-        'dist/release/style.css'
-      ]
+      compress: {
```
Comment (marktheunissen) [inline controller/web/Gruntfile.js:42]: Used the best practice guidelines from the plugin's home page

File: controller/web/Gruntfile.js [hunk 2]
```diff
@@ -13,101 +13,117 @@ module.exports = function(grunt) {
...
-        debugsass: false,
-        images: 'images',
-        fonts: 'fonts',
-        relativeassets: true
+        force: true,
+        imagesDir: 'images',
+        fontsDir: 'fonts',
+        relativeAssets: true
+      },
+      dev: {
```
Comment (marktheunissen) [inline controller/web/Gruntfile.js:64]: Just override specific options for 'dev' and 'prod' targets

File: controller/web/Gruntfile.js [hunk 3]
```diff
@@ -13,101 +13,117 @@ module.exports = function(grunt) {
...
     watch: {
       compass: {
-        files: ['Gruntfile.js', 'sass/**/*.scss'],
-        tasks: 'compass:dev compass:prod mincss'
+        files: ['sass/**/*.scss'],
+        tasks: ['compass:dev', 'compass:prod', 'cssmin']
       },
       requirejs: {
-        files: ['Gruntfile.js', 'js/src/**/*.js'],
-        tasks: 'jshint requirejs concat uglify'
```
Comment (marktheunissen) [inline controller/web/Gruntfile.js:103]: Uglify takes 50+ seconds, so don't do it when watching (debugging).

File: controller/web/Gruntfile.js [hunk 4]
```diff
@@ -6,108 +6,137 @@ module.exports = function(grunt) {
...
         scripturl: true
+      },
+      prod: {
+        files: {
+          src: ['Gruntfile.js', 'js/src/**/*.js']
+        }
+      },
+      dev: {
+        options: {
+          debug: true
```
Comment (marktheunissen) [inline controller/web/Gruntfile.js:19]: This allows us to use debugger; in source files for development.

**Comment by marktheunissen** [comment]
I have merged this into the course creation form as I need it to carry on. All this means is that this PR should land before that one, which should be easy to do.

**Comment by elliotttf** [comment]
Looks good to me!

**Pull request merged**